lib/portlayer/logging/logging.go

```diff
@@ -15,13 +15,123 @@
 package logging
 
 import (
+	"context"
 	"fmt"
+	"sync"
 
+	log "github.com/Sirupsen/logrus"
+
+	"github.com/vmware/govmomi/object"
 	"github.com/vmware/govmomi/vim25/types"
+	"github.com/vmware/vic/lib/portlayer/event/collector/vsphere"
+	"github.com/vmware/vic/lib/portlayer/event/events"
 	"github.com/vmware/vic/lib/portlayer/exec"
+	"github.com/vmware/vic/pkg/backoff"
 	"github.com/vmware/vic/pkg/trace"
 )
 
+var once sync.Once
+
+func Init(ctx context.Context) error {
+	once.Do(func() {
+		// Subscribe to vm events
+		exec.Config.EventManager.Subscribe(
```

Comment:
Tried it but failed because of the import cycles. Don't want to put the serial related code into the exec and exec can't use other portlayer subsystems because of the exec.Handle usage.